hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java

```diff
@@ -0,0 +1,505 @@
+                        cryptoCreate(TOKEN_TREASURY),
+                        cryptoCreate(SIGNER).balance(ONE_MILLION_HBARS),
+                        tokenCreate(FUNGIBLE_TOKEN)
+                                .tokenType(TokenType.FUNGIBLE_COMMON)
+                                .initialSupply(initialAmount)
+                                .treasury(TOKEN_TREASURY)
+                                .adminKey(TOKEN_TREASURY)
+                                .supplyKey(TOKEN_TREASURY)
+                                .exposingCreatedIdTo(idLit -> fungible.set(asToken(idLit))),
+                        uploadInitCode(HTS_CALLS),
+                        contractCreate(HTS_CALLS),
+                        uploadInitCode(MINT_CONTRACT),
+                        sourcing(() -> contractCreate(
+                                MINT_CONTRACT, HapiParserUtil.asHeadlongAddress(asAddress(fungible.get())))))
+                .when(withOpContext((spec, opLog) -> allRunFor(
+                        spec,
+                        contractCall(
+                                        HTS_CALLS,
+                                        "burnTokenCall",
+                                        HapiParserUtil.asHeadlongAddress(
+                                                asAddress(spec.registry().getTokenID(FUNGIBLE_TOKEN))),
+                                        BigInteger.valueOf(amountToBurn),
+                                        new long[0])
+                                .via(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
+                                .gas(GAS_TO_OFFER)
+                                .payingWith(SIGNER)
+                                .signedBy(SIGNER),
+                        // verify that the total supply of the tokens is not affected
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount),
+                        getTxnRecord(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
+                                .andAllChildRecords()
+                                .logged(),
+                        // Create a key with thresh 1/2 with sigs:  new ed25519 key, contractId of MINT_CONTRACT
+                        // contract. MINT_CONTRACT is used only as a "wrong" contract id
+                        newKeyNamed(TRESHOLD_KEY_WITH_SIGNER_KEY)
+                                .shape(TRESHOLD_KEY_SHAPE.signedWith(sigs(ON, MINT_CONTRACT))),
+                        // Update the signer of the transaction to have the threshold key with the wrong contract id
+                        cryptoUpdate(SIGNER).key(TRESHOLD_KEY_WITH_SIGNER_KEY),
+                        // Update the token's supply to have the threshold key with the wrong contract id
+                        tokenUpdate(FUNGIBLE_TOKEN).supplyKey(TRESHOLD_KEY_WITH_SIGNER_KEY),
+                        // Test Case 2: Signer paying and signing a token burn transaction, when the token
```

Comment:
In this test we don't have "Test Case 1:" as a comment and its confusing reading directly Test Case 2